src/utils.cpp

```diff
@@ -360,4 +361,328 @@ int mega_snprintf(char *s, size_t n, const char *format, ...)
 }
 #endif
 
+string * TLVstore::TLVrecordsToContainer(SymmCipher *key, encryptionmode_t mode)
+{    
+    // decide nonce/IV and auth. tag lengths based on the `mode`
+    unsigned ivlen = TLVstore::getIvlen(mode);
+    unsigned taglen = TLVstore::getTaglen(mode);
+    unsigned encMode = TLVstore::getMode(mode);
+
+    if (!ivlen || !taglen || !encMode)
+    {
+        return NULL;
+    }
+
+    // serialize the TLV records
+    string *container = TLVrecordsToContainer();
+
+    // generate IV array
+    byte *iv = new byte[ivlen];
+    PrnGen::genblock(iv, ivlen);
+
+    string cipherText;
+
+    // encrypt the bytes using the specified mode
+    if (encMode == 1)   // CCM or GCM_BROKEN (same than CCM)
```

Comment:
I'd prefer to use a named constant here, not a literal constant (e. g. use `CCM` or whichever it's in this code, and not `1`)